Yin/Resources/VideoViewController.swift

```diff
@@ -7,11 +7,31 @@ class VideoViewController: UIViewController {
 
     //objective c: has NSInteger, which is same as Int in swift, 32 or 64 bits depends on the processor.
     //objective c: int is from C, it is 32 bits
+    
+    // `open` is only needed when you want to be able to override the variable or function from other modules
     open var player : AVPlayer?
     var pageIndex:Int = 0
     var videoURI:URL
     init(videoURI:URL, pageIndex:Int) {
         //TODO: why in swift this is put it before super?
+        // You always need to init your own variables before calling super. Otherwise if super.init() calls any instance method that you have overriden then and used one of your new instance variables, that variable would be uninitialized.
```

Comment:
In Objective c 

```
- (instanceTyype)initWithName:(NSString *name) {
       if self = [super init] {
          _name = name;
      }
     return self;
}
```

So here you call super first and then init with variable. 

When you say always, do you mean in swift world, it is always ? or just in general for everything ?